src/NuGetSdkResolver/NuGetSdkResolverBase.cs

```diff
@@ -76,8 +99,7 @@ static NuGetSdkResolverBase()
             AssemblyName assemblyName)
         {
 #endif
-            // Only load NuGet.* and Newtonsoft.Json assemblies if found
-            if (assemblyName.Name.StartsWith("NuGet.", StringComparison.OrdinalIgnoreCase) || assemblyName.Name.Equals("Newtonsoft.Json", StringComparison.OrdinalIgnoreCase))
+            if (NuGetAssemblies.Contains(assemblyName.Name))
```

Comment:
Is it worth using an ordinal case ignoring comparer?

Reply:
Good point, I'll add it to the `HashSet<T>`